src/build-remote/build-remote.cc

```diff
@@ -130,6 +130,38 @@ static int main_build_remote(int argc, char * * argv)
                 for (auto & m : machines) {
                     debug("considering building on remote machine '%s'", m.storeUri);
 
+                    if (std::find(m.systemTypes.begin(), m.systemTypes.end(), "*") != m.systemTypes.end() ||
+                        std::find(m.supportedFeatures.begin(), m.supportedFeatures.end(), "*") != m.supportedFeatures.end()) {
+                        // wildcard is used, so we need to ask the machine what it supports
+                        try {
+                            Activity act(*logger, lvlTalkative, actUnknown, fmt("connecting to '%s'", m.storeUri));
+
+                            auto store = m.openStore();
+
+                            if (hasPrefix(m.storeUri, "ssh://"))
+                                throw Error("Cannot use wildcards with legacy ssh store. Instead use the ssh-ng:// protocol.");
```

Comment:
I think this will not just fail for `ssh` stores, but also for any store other than `ssh-ng` (`/foo/bar`, `unix:/other/daemon/socket`, etc..) as their `systemTypes` field will never be set

Reply:
Maybe the constructor of `LocalFSStore` could initialize these with the same values that are used in the new daemon code

Reply:
We can cover this in a test, but I think unix: (UDSRemoteStore) will work fine because it uses RemoteStore. We should be able to support anything that uses the worker protocol. We could support the ssh protocol too, but I got the impression it's meant to be gradually phased out.

Reply:
Oh yes of course, `UDSRemoteStore` will work just fine.

I think the point still stand about the `LocalStore` though, esp. now that it’s possible to do something like `nix build --store https://my.binary.cache --builders 'local …'`